public/app/models/finding_aid_pdf.rb

```diff
@@ -65,9 +81,17 @@ def source_file
         end
       }
 
-      writer.write(renderer.render_to_string partial: 'toc', layout: false, :locals => {:resource => @resource, :has_children => has_children, :ordered_aos => toc_aos})
+      begin
+        writer.write(renderer.render_to_string partial: 'toc', layout: false, :locals => {:resource => @resource, :has_children => has_children, :ordered_aos => toc_aos})
+      rescue => e
+        raise PDFRenderErrorTOC, "#{e.class};#{e.message}"
+      end
 
-      writer.write(renderer.render_to_string partial: 'resource', layout: false, :locals => {:record => @resource, :has_children => has_children})
+      begin
+        writer.write(renderer.render_to_string partial: 'resource', layout: false, :locals => {:record => @resource, :has_children => has_children})
+      rescue => e
+        raise PDFRenderErrorResource, "#{e.class};#{e.message}"
```

Comment:
Since we know the resource details, we could include the title and / or id of the resource in the error message that the user sees